WcaOnRails/app/models/competition.rb

```diff
@@ -331,6 +331,22 @@ def warnings_for(user)
       unless has_any_round_per_event? && schedule_includes_rounds?
         warnings[:schedule] = I18n.t('competitions.messages.schedule_must_match_rounds')
       end
+
+      if championship_warnings.any?
+        warnings = championship_warnings.merge(warnings)
+      end
+    end
+
+    warnings
+  end
+
+  def championship_warnings
+    warnings = {}
+    for championship in self.championships
+      existing_championship = Championship.joins(:competition).merge(Competition.visible).where(championship_type: championship.championship_type, Competitions: {year: self.year})
```

Comment:
You can use `exists?` instead of `where` to select a boolean directly from the database, instead of the whole record =)